integration/tests/bft/wintermute/suite.go

```diff
@@ -120,91 +90,30 @@ func (s *Suite) SetupSuite() {
+	s.NodeConfigs = append(s.NodeConfigs, exe2Config)
 
 	// honest EN
 	s.honestENId = unittest.IdentifierFixture()
 	exe3Config := testnet.NewNodeConfig(flow.RoleExecution,
 		testnet.WithID(s.honestENId),
 		testnet.WithLogLevel(zerolog.ErrorLevel))
-	s.nodeConfigs = append(s.nodeConfigs, exe3Config)
-
-	// generates two collection node
-	coll1Config := testnet.NewNodeConfig(flow.RoleCollection,
-		testnet.WithLogLevel(zerolog.FatalLevel),
-	)
-	coll2Config := testnet.NewNodeConfig(flow.RoleCollection,
-		testnet.WithLogLevel(zerolog.FatalLevel),
-	)
-	s.nodeConfigs = append(s.nodeConfigs, coll1Config, coll2Config)
+	s.NodeConfigs = append(s.NodeConfigs, exe3Config)
 
-	// Ghost Node
-	// the ghost node's objective is to observe the messages exchanged on the
-	// system and decide to terminate the test.
-	// By definition, ghost node is subscribed to all channels.
-	s.ghostID = unittest.IdentifierFixture()
-	ghostConfig := testnet.NewNodeConfig(flow.RoleExecution,
-		testnet.WithID(s.ghostID),
-		testnet.AsGhost(),
-		testnet.WithLogLevel(zerolog.FatalLevel))
-	s.nodeConfigs = append(s.nodeConfigs, ghostConfig)
-
-	// generates, initializes, and starts the Flow network
-	netConfig := testnet.NewNetworkConfig(
+	s.BaseSuite.StartCorruptedNetwork(
 		"wintermute_tests",
-		s.nodeConfigs,
-		// set long staking phase to avoid QC/DKG transactions during test run
-		testnet.WithViewsInStakingAuction(10_000),
-		testnet.WithViewsInEpoch(100_000),
+		10_000,
+		100_000,
+		func() insecure.AttackOrchestrator {
+			s.Orchestrator = wintermute.NewOrchestrator(s.Log, s.Net.CorruptedIdentities().NodeIDs(), s.Net.Identities())
+			return s.Orchestrator
+		},
 	)
-
-	s.net = testnet.PrepareFlowNetwork(s.T(), netConfig, flow.BftTestnet)
-
-	ctx, cancel := context.WithCancel(context.Background())
-	s.cancel = cancel
-	s.net.Start(ctx)
-
-	s.Orchestrator = wintermute.NewOrchestrator(s.log, s.net.CorruptedIdentities().NodeIDs(), s.net.Identities())
-
-	// start orchestrator network
-	codec := unittest.NetworkCodec()
-	connector := orchestrator.NewCorruptedConnector(s.log, s.net.CorruptedIdentities(), s.net.CorruptedPortMapping)
-	orchestratorNetwork, err := orchestrator.NewOrchestratorNetwork(s.log,
-		codec,
-		s.Orchestrator,
-		connector,
-		s.net.CorruptedIdentities())
-	require.NoError(s.T(), err)
-	s.orchestratorNetwork = orchestratorNetwork
-
-	attackCtx, errChan := irrecoverable.WithSignaler(ctx)
-	go func() {
-		select {
-		case err := <-errChan:
-			s.T().Error("orchestratorNetwork startup encountered fatal error", err)
-		case <-ctx.Done():
-			return
-		}
-	}()
-
-	orchestratorNetwork.Start(attackCtx)
-	unittest.RequireCloseBefore(s.T(), orchestratorNetwork.Ready(), 1*time.Second, "could not start orchestrator network on time")
-
-	// starts tracking blocks by the ghost node
-	s.Track(s.T(), ctx, s.Ghost())
-}
-
-// TearDownSuite tears down the test network of Flow as well as the BFT testing orchestrator network.
```

Comment:
can you add to the PR description how the `TearDownSuite` from each type of BFT test was extracted to a common base / common implementation? Easier to follow the changes that way.

Reply:
TearDownSuite simply removes the network each integration tests suite does it. I added a list of the reusable methods shared among each BFT test. The only difference between all of them is the number and type of corrupted nodes each test uses.